src/login7-payload.js

```diff
@@ -237,7 +256,12 @@ module.exports = class Login7Payload {
     offset = fixedData.writeUInt16LE(dataOffset, offset);
 
     // (cchUnused / cbExtension): 2-byte
-    offset = fixedData.writeUInt16LE(0, offset);
+    const extensions = this.buildFeatureExt();
+    offset = fixedData.writeUInt16LE(4, offset);
+    const extensionOffset = new Buffer(4);
+    extensionOffset.writeUInt32LE(dataOffset += 4, 0);
```

Comment:
also, update Buffer() to Buffer.alloc()/Buffer.from() for all 3 occurrences in this file

Reply:
Done. 👍